tests/MongoDB.Driver.Tests/UnifiedTestOperations/UnifiedCreateEntitiesOperation.cs

```
@@ -0,0 +1,69 @@
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using MongoDB.Bson;
+
+namespace MongoDB.Driver.Tests.UnifiedTestOperations
+{
+    public sealed class UnifiedCreateEntitiesOperation : IUnifiedEntityTestOperation
+    {
+        private readonly BsonArray _entitiesArray;
+        private readonly UnifiedEntityMap _entityMap;
+
+        public UnifiedCreateEntitiesOperation(BsonArray entitiesArray, UnifiedEntityMap entityMap)
+        {
+            _entitiesArray = entitiesArray;
+            _entityMap = entityMap;
+        }
+
+        public OperationResult Execute(CancellationToken cancellationToken)
+        {
+            try
+            {
+                var newEntityMap = new UnifiedEntityMapBuilder(null, _entityMap.LoggerFactory).Build(_entitiesArray);
+                _entityMap.AddEntities(newEntityMap);
+
+                return OperationResult.FromResult(null);
+            }
+            catch (Exception exception)
+            {
+                return OperationResult.FromException(exception);
+            }
+        }
+
+        public Task<OperationResult> ExecuteAsync(CancellationToken cancellationToken) =>
+           Task.FromResult(Execute(cancellationToken));
+    }
+
+    public class UnifiedCreateEntitiesOperationBuilder
+    {
+        private readonly UnifiedEntityMap _entityMap;
+
+        public UnifiedCreateEntitiesOperationBuilder(UnifiedEntityMap entityMap)
+        {
+            _entityMap = entityMap;
+        }
+
+        public UnifiedCreateEntitiesOperation Build(BsonDocument arguments)
+        {
+            var entities = arguments["entities"].AsBsonArray;
```

we usually add validation on arguments. In your case, validation that there is no any argument to make sure that we don't silently ignore it

---

Done.